### ISSUE TITLE
Json in rest-assured request is printed as Map 

### ISSUE THREAD

**Issue by dsmt** [serenity-bdd/serenity-core#197]
It would be nice to print json in request body as json, not as Map. It will definitely facilitate reproducing defects if something went wrong with REST-request.

![](http://take.ms/INtlt)

**Comment by YamStranger** [comment]
![json_map](https://cloud.githubusercontent.com/assets/1667699/11351520/cf080222-923f-11e5-99b5-6376586794c7.JPG)

**Comment by dsmt** [comment]
Well, to be more explicit I reproduced this issue using **junit-webtests** demo.
I added another step to **PetStoreSteps** which adds content according to [rest-assured manual](https://code.google.com/p/rest-assured/wiki/Usage#Create_JSON_from_a):

    @Step
    public void when_i_add_the_pet_to_the_store_with_json_map(Pet pet) {
        this.pet = pet;
        int id = Math.abs(new Random().nextInt());
        Map<String, Object> jsonAsMap = new HashMap<>();
        jsonAsMap.put("id", id);
        jsonAsMap.put("name", pet.getName());
        jsonAsMap.put("status", pet.getStatus());
        jsonAsMap.put("photoUrls", new ArrayList<>(Arrays.asList()));

        rest().given().contentType("application/json")
                .content(jsonAsMap).post("http://petstore.swagger.io/v2/pet");

        this.pet.setId(id);
    }

It adds the json as when_i_add_the_pet_to_the_store does:

    @Step
    public void when_i_add_the_pet_to_the_store(Pet pet) {
        this.pet = pet;
        int id = Math.abs(new Random().nextInt());
        this.jsonPet = "{\"id\": " + id + " , \"name\": \""
                + pet.getName() + "\", \"photoUrls\": [], \"status\": \""
                + pet.getStatus() + "\"}";

        rest().given().contentType("application/json")
                .content(jsonPet).post("http://petstore.swagger.io/v2/pet");

        this.pet.setId(id);
    }

Then I added new step to the test:

    @Test
    public void shouldBeAbleDeletePetsFromAStore() {

        Pet fido = new Pet("available","fido");

        petStore.when_i_add_the_pet_to_the_store(fido);
        
        petStore.when_i_add_the_pet_to_the_store_with_json_map(fido);

        petStore.when_i_delete_the_pet();

        petStore.the_pets_should_be_not_available();
    }

Here are my Serenity logs:

![](http://take.ms/Nc14Y)

![](http://take.ms/3SGO3)

**Comment by YamStranger** [comment]
Hello @dsmt!

This issue should be fixed under [1.1.22-rc.11](https://github.com/serenity-bdd/serenity-core/releases/tag/v1.1.22-rc.11), please check it. 

If you feel that this issue already fixed and cant be reproduced - please close this issue. 

Thanks a lot!!

**Comment by dsmt** [comment]
Hi, @YamStranger. I checked this issue, unfortunately current fix didn't help. I have found two new defects:

* Request body with Map is not even appear in report
* Body with XML (as plane text and xml in content-type) throws exception

![](http://take.ms/loleX)

![](http://take.ms/2lxAc)

**Comment by YamStranger** [comment]
@dsmt I will check it now, thanks

**Comment by YamStranger** [comment]
@dsmt You are right, I reverted changes for this issue, will find what gone wrong - and fix this issue. 

Thanks for such fast response!!

**Comment by YamStranger** [comment]
depends on #355

**Comment by YamStranger** [comment]
Hello @pnsantos 
You issue is fixed and can be checked in serenity-core 1.1.31. 

Now for printing body we use internals of rest-assured - so you can be sure that printed body is the same, as was included in request/response. 

To check your issue I created class: 
```
public class ASimpleStepWithExpectAndContentAsMap {

    List<Pet> pets;

    public void givenIHaveFollowingPets(Map<String, String> values) {

        this.pets = Lists.newArrayList();

        for (Map.Entry<String, String> petRow : values.entrySet()) {
            this.pets.add(new Pet(petRow.getValue(), petRow.getKey()));
        }
    }

    public void whenIAddThePetToTheStore() {
        for (Pet pet : pets) {
            int id = Math.abs(new Random().nextInt());
            Map<String, Object> jsonAsMap = new HashMap<>();
            jsonAsMap.put("id", id);
            jsonAsMap.put("name", pet.getName());
            jsonAsMap.put("status", pet.getStatus());
            jsonAsMap.put("photoUrls", new ArrayList<>(Arrays.asList()));

            given().contentType("application/json")
                    .content(jsonAsMap).log().body().post("http://petstore.swagger.io/v2/pet");

            pet.setId(id);

        }
    }

    public void thenPetShouldBeAvailable() {
        for (Pet pet : pets) {
            expect().that().statusCode(200)
                    .and().body("name", equalTo(pet.getName())).when()
                    .get("http://petstore.swagger.io/v2/pet/{id}", pet.getId());
        }
    }
}
```

and after build here is report:
![before2](https://cloud.githubusercontent.com/assets/1667699/14665216/12fa8018-06da-11e6-889c-ed302d417f0e.png)
![before1](https://cloud.githubusercontent.com/assets/1667699/14665217/12fc40c4-06da-11e6-9120-92da5417fff5.png)


please close this ticket

Thanks a lot!

**Comment by YamStranger** [comment]
@dsmt I updated code to make repost more user-friendly:
```

    @Step
    public void whenIAddThePetToTheStore() {
        for (Pet pet : pets) {
            int id = Math.abs(new Random().nextInt());
            Map<String, Object> jsonAsMap = new HashMap<>();
            jsonAsMap.put("id", id);
            jsonAsMap.put("name", pet.getName());
            jsonAsMap.put("status", pet.getStatus());
            jsonAsMap.put("photoUrls", new ArrayList<>(Arrays.asList()));

            given().contentType("application/json")
                    .content(jsonAsMap).log().body()
                    .baseUri("http://petstore.swagger.io")
                    .basePath("v2/pet")
            .when().post();

            pet.setId(id);

        }
    }
```

![before3](https://cloud.githubusercontent.com/assets/1667699/14665567/bb9f1a1a-06dc-11e6-9cda-012d05e16d92.png)
